memory_profiler.py

```diff
@@ -736,10 +754,16 @@ def trace_memory_usage(self, frame, event, arg):
                 self.prevlines.append(frame.f_lineno)
             elif event == 'line':
                 # trace needs current line and previous line
-                self.code_map.trace(frame.f_code, self.prevlines[-1], self.prev_lineno)
-                # saving previous line
-                self.prev_lineno = self.prevlines[-1]
-                self.prevlines[-1] = frame.f_lineno
+                _run = True
+                if self.backend == 'guppy':
+                    if self.prevlines[-1] == self.prev_lineno:
+                        # the reason for this is because for code like [i for i in range(0, 1000000)] guppy will be extremely slow
+                        _run = False
```

Comment:
this can cause issues if you want to measure the memory of a side effect eg
```
[foo() for i in range(0, 500)]
def foo():
     make_big_array = random.randint(0, int(1e9))
```
i haven't tested it.
although it will give correct results for this case https://github.com/pythonprofilers/memory_profiler/issues/236#issuecomment-549044946